### PR TITLE
feat: add disabled prop to nav item #376

### LINKED ISSUE #376: Add disabled prop to nav_item
Allow disabling a nav item.

Steps to follow:
1. Setup the project.
2. Find `nav.tsx`.
3. Implement needed feature.
4. run `make generate` from project root - this generates python API.
5. Use your newly implemented prop to showcase its capabilities.
6. Add unit tests.

### PR DESCRIPTION
**Pull request by srini-x**
closes #376 

1. - [x] Setup the project.
2. - [x] Find nav.tsx.
3. - [x] Implement needed feature.
4. - [x] run make generate from project root - this generates python API.
5. - [x] Use your newly implemented prop to showcase its capabilities.
6. - [x] Add unit tests.

### CODE REVIEW

File: ui/src/nav.tsx [hunk 1]
```diff
@@ -25,6 +25,8 @@ export interface NavItem {
   label: S
   /** An optional icon to display next to the label. */
   icon?: S
+  /** True if the nav_item should be disabled. */
```
Comment (lo5) [inline ui/src/nav.tsx:28]:
```suggestion
  /** True if this item should be disabled. */
```

File: py/examples/nav.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ async def serve(q: Q):
                     ui.nav_item(name='#menu/spam', label='Spam'),
                     ui.nav_item(name='#menu/ham', label='Ham'),
                     ui.nav_item(name='#menu/eggs', label='Eggs'),
+                    ui.nav_item(name='#menu/disabled', label='Disabled', disabled=True),
```
Comment (lo5) [inline py/examples/nav.py:24]:
```suggestion
                    ui.nav_item(name='#menu/toast', label='Toast', disabled=True),
```

**Review by lo5 — CHANGES_REQUESTED** [review]
Thanks @srini-x - just minor comments on docs.

**Comment by srini-x** [comment]
@lo5 I updated the changes using the interface provided by GitHub on this page. Is it better to update on my computer, squash and push?

**Review by lo5 — APPROVED** [review]
Thanks @srini-x!

**Pull request merged**